src/rez/solver.py

```diff
@@ -907,7 +931,8 @@ def __str__(self):
             s = "[%s==%s%s]" % (self.package_name, str(variant.version), s_idx)
         elif nversions == 1:
             entry = self.entries[0]
-            indexes = sorted([x.index for x in entry.variants])
+            # we expect all variants to have a non-None index, but filter to satisfy mypy
+            indexes = sorted([x.index for x in entry.variants if x.index is not None])
```

Comment:
This could have a performance impact.

Reply:
I can change this to use a `typing.cast` to change the type without any runtime impact, as long as we're confident that the index is never None.